*** bioflex/bioflex/src/org/jbei/bio/sequence/common/StrandedAnnotation.as.orig	2011-12-20 14:54:16.888301622 -0800
--- bioflex/bioflex/src/org/jbei/bio/sequence/common/StrandedAnnotation.as	2011-12-20 13:41:51.502754017 -0800
***************
*** 1,7 ****
  package org.jbei.bio.sequence.common
  {
      [RemoteClass(alias="org.jbei.bio.sequence.common.StrandedAnnotation")]
!     
      /**
      * Annotation with defined read direction.
      * 
--- 1,7 ----
  package org.jbei.bio.sequence.common
  {
      [RemoteClass(alias="org.jbei.bio.sequence.common.StrandedAnnotation")]
! 
      /**
      * Annotation with defined read direction.
      * 
***************
*** 21,30 ****
          * @param end Annotation end
          * @param strand Strand type. @see org.jbei.bio.sequence.common.StrandType
          */
!         public function StrandedAnnotation(start:int=0, end:int=0, strand:int = StrandType.FORWARD)
          {
              super(start, end);
              
              _strand = strand;
          }
          
--- 21,39 ----
          * @param end Annotation end
          * @param strand Strand type. @see org.jbei.bio.sequence.common.StrandType
          */
! 
!         // Marc Juul changed: strand:int was not a compile-time constant (gave compiler error in Flex 3)
! 
!         public function StrandedAnnotation(start:int=0, end:int=0, strand:int = -1)
          {
              super(start, end);
              
+            // Marc Juul changed: strand:int was not a compile-time constant (gave compiler error in Flex 3)
+ 
+             if(strand == -1) {
+                 strand = StrandType.FORWARD;
+             }
+ 
              _strand = strand;
          }
          
